ubuntu22.04/nvidia-driver

```diff
@@ -195,6 +195,34 @@ _assert_nvswitch_system() {
     return 0
 }
 
+_assert_nvlink5_system() (
+    for dir in /sys/class/infiniband/*/device; do
+        # Define the path to the VPD file
+        vpd_file="$dir/vpd"
+
+        # Check if the VPD file exists
+        if [ -f "$vpd_file" ]; then
+            # Search for 'SW_MNG' in the VPD file
+            if grep -q "SW_MNG" "$vpd_file"; then
+                echo "Detected NVLink5+ system"
+                return 0
+            fi
+        fi
+    done
+    return 1
+)
+
+_ensure_nvlink5_prerequisites() (
+    until  lsmod | grep mlx5_core > /dev/null 2>&1 && lsmod | grep ib_umad > /dev/null 2>&1;
+    do
+        echo "waiting for the mlx5_core and ib_umad kernel modules to be loaded"
+        sleep 10
+    done
+
+    echo "Installing the infiniband-diags package"
+    apt-get -qq install --no-install-recommends infiniband-diags > /dev/null
```

Comment:
Why is this package required?

Reply:
the `fabricmanager` runs the `ibstat` command to fetch the guids of the NVLink5 switches. ibstat is a part of the `infiniband-diags` pakage

Reply:
If `infiniband-diags` is small in size, should we consider installing it at build time?

Reply:
Sure, it seems small enough

Reply:
Done